src/get-started/java-binding.md

```diff
@@ -491,75 +493,201 @@ as specified in [How to Run a Service section](#how-to-run-a-service).
-      }
+Default TestKit can be instantiated with a single validator as an emulated node,
+single service and without Time Oracle in the following way:
 
-      @After
-      public void tearDown() throws Exception {
-        cleaner.close();
-        db.close();
-      }
+```java
+try (TestKit testKit = TestKit.forService(MyServiceModule.class)) {
+  // Test logic
+}
+```
 
-      @Test
-      public void testCreatingSnapshot() {
-        Snapshot snapshot = db.createSnapshot(cleaner);
-        // Use the Snapshot to test the schema.
-      }
+Or using a builder if different configuration is needed:
 
-      @Test
-      public void testMergingSomeChanges() {
-        Fork fork = db.createFork(cleaner);
-        // Make some changes to Fork.
-        // …
-
-        // Merge the changes made to the database Fork into the database.
-        db.merge(fork);
-
-        // Create a Snapshot of the database state. Must include the changes
-        // made above.
-        Snapshot snapshot = db.createSnapshot(cleaner);
-        // Use the Snapshot to test the schema/transaction
-        // with a certain database state.
-        // …
-      }
-    }
-    ```
+```java
+try (TestKit testKit = TestKit.builder()
+    .withServices(MyServiceModule.class, MyServiceModule2.class)
+    .withValidators(2)
+    .build()) {
+  // Test logic
+}
+```
 
-### Transactions
+### Transactions testing
 
-To test transactions execution, you can use `MemoryDb`, as in the previous
-section.
+TestKit allows testing transaction execution by submitting blocks with given
+[transaction messages][transactions-messages]:
+
+```java
+try (TestKit testKit = TestKit.forService(MyServiceModule.class)) {
+  // Construct some transaction to be executed
+  TransactionMessage message = constructTransactionMessage();
+  // Commit block with this transaction
+  Block block = testKit.createBlockWithTransactions(message);
+  // Check the resulting block or blockchain state
+}
+```
+
+TestKit also allows creating blocks that contain all current [in-pool][in-pool]
+transactions:
+
+```java
+try (TestKit testKit = TestKit.forService(MyServiceModule.class)) {
+  // Put the transaction into TestKit transaction pool
+  MyService service = testKit.getService(MyService.SERVICE_ID, MyService.class);
+
+  TransactionMessage message = constructTransactionMessage();
+  RawTransaction rawTransaction = RawTransaction.fromMessage(message);
+  service.getNode().submitTransaction(rawTransaction);
 
-### Read Requests
+  // This block will contain transaction submitted above
+  Block block = testKit.createBlock();
+  // Check the resulting block or blockchain state
+}
+```
+
+TestKit provides [`getTransactionPool()`][testkit-get-pool] and
+[`findTransactionsInPool(Predicate<TransactionMessage> predicate)`][testkit-find-in-pool]
+methods to inspect transaction pool. It is useful when there is a need to verify
+transactions submitted in `afterCommit` method, as those are put into the pool.
```

Comment:
I'd clarify as
"transactions that the service submitted itself (e.g., in `afterCommit` method), …"